docs: include generics FAQ (#186)
felangel/mocktail: packages/mocktail/README.md

final person = MockPerson();
when(() => person.sleep()).thenAnswer((_) async {});
```

#### Why is my method throwing a `TypeError` when stubbing using `any()`?

[Relevant Issue](https://github.com/felangel/mocktail/issues/162)

> By default when a class extends `Mock` any unstubbed methods return `null`. When stubbing using `any()` the type must be inferable. However, when a method has a generic type argument it might not be able to infer the type and the method would act as if it was unstubbed.

Reviewer: ```suggestion
By default when a class extends `Mock` any unstubbed methods return `null`. When stubbing using `any()` the type must be inferable. However, when a method has a generic type argument it might not be able to infer the type and as a result, the generic would fallback to `dynamic` causing the method to act as if it was unstubbed.
```

For example, take the following class and its method:

```dart
class MockPerson extends Mock implements Person {}

class Person {
  T doSomething<T>(T value) => value;
}
```

> Stubbing `doSomething` won't be able to infer the type when doing:

Reviewer: ```suggestion
The following stub will be equivalent to calling `doSomething<dynamic>(...)`:
```

```dart

> // The type of `any<T>()` can't be inferred.

Reviewer: ```suggestion
// The type `T` of `any<T>()` is inferred to be `dynamic`.
```

For example, take the following class and its method:

```dart
class Cache {
  bool set<T>(String key, T value) {
    return true;
  }
}
```

The following stub will be equivalent to calling `set<dynamic>(...)`:

```dart
// The type `T` of `any<T>()` is inferred to be `dynamic`.

> when(() => cache.set(any(), any())).thenAnswer((_) => true);

Reviewer: ```suggestion
when(() => cache.set(any(), any())).thenReturn((_) => true);
```

```

To address this, we must explicitly stub `set` with a type:

```dart
final cache = MockCache();

> when(() => cache.set<int>(any(), any())).thenAnswer((_) => true);

Reviewer: ```suggestion
when(() => cache.set<int>(any(), any())).thenReturn((_) => true);
```

cache.set<int>('key', 1);
verify(() => cache.set<int>(any(), any())).called(1);
```

The type doesn't need to be applied to `set<T>()`, any explicit type that allows `any<T>()` infer its type will allow the method to be stubbed for that type:

```dart

> when(() => cache.set(any(), any<int>())).thenAnswer((_) => true);

Reviewer: ```suggestion
when(() => cache.set(any(), any<int>())).thenReturn((_) => true);
```
